crates/l2/proposer/prover_server.rs

```diff
@@ -392,27 +403,33 @@ impl ProverServer {
         Ok(())
     }
 
-    fn create_prover_input(&self, block_number: u64) -> Result<ProverInputData, String> {
+    fn create_prover_input(&self, block_number: u64) -> Result<ProverInputData, ProverServerError> {
         let header = self
             .store
             .get_block_header(block_number)
-            .map_err(|err| err.to_string())?
-            .ok_or("block header not found")?;
+            .map_err(StoreError::from)?
```

Comment:
It is no longer necessary to cast to `StoreError` here.

Reply:
fixed